### ISSUE TITLE
How do exact redirects interact with other redirects?

### ISSUE THREAD

**Issue by astrojuanlu** [readthedocs/readthedocs.org#9988]
## Details

* Read the Docs project URL: https://readthedocs.org/projects/kedro/

We have some questions about redirects - I was hesitant whether to send a private support request (as encouraged by the issue form), but since one could consider this a documentation bug, I went ahead and opened an issue instead.

_Comes from https://stackoverflow.com/q/75321853/554319, opened by my colleague @stichbury_

I currently have docs on RTD (and have had them there for 3 years). We have moved pages around as versions changed, so have a set of about a hundred exact redirects, something like this:

```
/en/stable/introduction/intro.html -> /en/stable/introduction/setup-guide.html
```

Now I have a custom domain and want to point everything with a canonical link over there.

What will happen to those redirects? Are they applied first so someone with a bookmark to `intro.html` is pointed to `setup-guide.html` over on `project.readthedocs.io` and then the redirect to `myprojectname.com/docs/` applied?

That is, will this happen?

```
/en/stable/introduction/intro.html --> myprojectname.com/docs/en/stable/introduction/setup-guide.html
```

Basically we're too scared we'll break the whole edifice and end world order.

**Comment by humitos** [comment]
This is a good question.

I'll have to double check with the code, but I'd say that people will be redirected twice:

1. project.readthedocs.io/path/ -> custom domain/path/
2. /path/ -> /new-path/

First redirect is called "system redirect" in the source code and the second one "user redirect"

**Comment by ANA885874** [comment]
Ok

**Comment by astrojuanlu** [comment]
Thanks a lot! This is really helpful. I think the conclusion is that adding a custom domain redirect should not break anything 😄 and in any case, seems like something that can be easily rolled back if the world crumbles.

cc @tynandebold @stichbury 

Closing this for now!

**Comment by stichbury** [comment]
Thanks @humitos and @astrojuanlu -- That's good to know and I think we have to take a deep breath and try it. I appreciate the ⭐ ⭐ ⭐ ⭐ ⭐  help! Thanks again.

**Comment by astrojuanlu** [comment]
FTR, we launched today `docs.kedro.org` and everything worked like a charm 💯

**Comment by humitos** [comment]
That's excellent news! Thanks for the feedback and sharing it back here 💪🏼
